pynucastro/rates/rate.py

```diff
@@ -1045,3 +1052,122 @@ def __lt__(self, other):
 
     def __eq__(self, other):
         return self.forward == other.forward and self.reverse == other.reverse
+
+
+class ApproximateRate(Rate):
+
+    def __init__(self, primary_rate, secondary_rates,
+                 primary_reverse, secondary_reverse, is_reverse=False, approx_type="ap_pg"):
+        """the primary rate has the same reactants and products and the final
+        approximate rate would have.  The secondary rates are ordered such that
+        together they would give the same sequence"""
+
+        self.primary_rate = primary_rate
+        self.secondary_rates = secondary_rates
+
+        self.primary_reverse = primary_reverse
+        self.secondary_reverse = secondary_reverse
+
+        self.is_reverse = is_reverse
+
+        self.approx_type = approx_type
+
+        if self.approx_type == "ap_pg":
+
+            # an ap_pg approximate rate combines A(a,g)B and A(a,p)X(p,g)B into a
+            # single effective rate by assuming proton equilibrium.
+
+            assert len(secondary_rates) == 2
+
+            # make sure that the primary forward rate makes sense
+            # this should be A(a,g)B
+
+            assert Nucleus("he4") in self.primary_rate.reactants and len(self.primary_rate.products) == 1
+
+            # we are going to define the product A and reactant B from this reaction
+
+            self.primary_reactant = sorted(self.primary_rate.reactants)[-1]
```

Comment:
`max(self.primary_rate.reactants)` is a bit more idiomatic

Reply:
agreed.

Reply:
I've replaced 3 of these